### ISSUE TITLE
Issue with RGB colourspace limits calculation.

### ISSUE THREAD

**Issue by ramparvathaneni** [colour-science/colour#835]
Hello,

I am currently using this package in my personal project. I am using the data in "ECI RGB v2" color space. The limits of this color space in LAB is given as 
 [[   0.          100.        ]
 [-146.82910546  102.01761727]
 [-102.83204539  104.92750053]]
by using print(colour.RGB_colourspace_limits(colour.RGB_COLOURSPACES['ECI RGB v2'])).
During the conversion of my data to LAB space, I have noticed that the B values are jumping out of the limits. That is B values are going beyond the 104.927, which I find strange. With the help of a color physicist involved in my project, we have found the bug in the code and fixed it.

After fixing the bug, these are the new results.
Colour space "ECI RGB v2":
White point "D50" XYZ=(0.9643, 1.0000, 0.8251)
xy=(0.3457, 0.3585)

Limits
L*:  0.0000, 100.0000)
a*:  -137.4806, 99.2890)
b*:  -108.8913, 109.2346)

I would like to request you to reflect on this issue and fix it if you find it as a correct calculation.

Please find the notes provided by him and also the code part:
A very nice resource about color science is the booklet/pdf by Konica Minolta: https://www.konicaminolta.com/instruments/knowledge/color/pdf/color_communication.pdf#page=55
On page 55 the equations for calculating CIE 1976  𝐿∗𝑎∗𝑏∗  are given.

(These equations are only valid for ratios above  (24/116)3, but that is not my point here.)
Where the normalization constants  (𝑋𝑛,𝑌𝑛,𝑍𝑛)  are the tristimulus values for a perfectly white surface under a given illuminant, and  (𝑋,𝑌,𝑍)  is are the CIE tristimulus for either the CIE 1931  2∘  Standard Observer (or the CIE 1964  10∘  Supplementary Standard Observer).

This definition with white point normalization  (𝑋𝑛,𝑌𝑛,𝑍𝑛)  implies that a transformation of RGB (via XYZ) into  𝐿∗𝑎∗𝑏∗  coordinates depends on the illuminant white point! It follows from the equations that form a perfectly white surface all fractions are unity and therefore the  𝐿∗𝑎∗𝑏∗  coordinate should be  (100,0,0). 

import colour 
import numpy as np 

def get_matrix_RGB_to_XYZ(cs_name='ECI RGB v2'): 
    '''Get RGB to XYZ transformation matrix from colour space name *cs_name*  
    and report white point name. 
    
    Returns: transformation matrix *M*'''
    
    cs_list = list(dict(colour.RGB_COLOURSPACES).keys())
    
    assert cs_name in cs_list, f'Colourspace not found. Choose one of: \n {cs_list}' 
    
    cs = colour.RGB_COLOURSPACES[cs_name] 
    
    M = cs.matrix_RGB_to_XYZ 
    Xw, Yw, Zw = M @ [1, 1, 1] 
    xw = Xw / (Xw + Yw + Zw)
    yw = Yw / (Xw + Yw + Zw)
    
    print(f'Colour space "{cs_name}":') 
    print(f'White point "{cs.whitepoint_name}" XYZ=({Xw:.4f}, {Yw:.4f}, {Zw:.4f})')
    print(f'xy=({xw:.4f}, {yw:.4f})')
    
    return M  
        

def my_limits(cs_name): 
    '''Calculate limets via my RGB conversion code and colour.XYZ_to_Lab.'''
      
    # 8 corners of the RGB cube 
    RGB_corners = [[0, 0, 0], [1, 0, 0], [0, 1, 0], [0, 0, 1], 
                   [1, 1, 0], [1, 0, 1], [0, 1, 1], [1, 1, 1]]

    RGB_white = [1, 1, 1] 
    M = get_matrix_RGB_to_XYZ(cs_name) 

    XYZ_white = M @ RGB_white  
    white_xy = XYZ_white[0:2] / np.sum(XYZ_white)
    white_xy

    Lab_corners = []
    for RGB_c in RGB_corners: 
        RGB_c = np.array(RGB_c)
        XYZ_c = M @ RGB_c
        Lab_c = colour.XYZ_to_Lab(XYZ_c, illuminant=white_xy)

        Lab_corners.append(Lab_c)

    Lab_corners = np.array(Lab_corners)

    L_max = Lab_corners[:, 0].max()
    L_min = Lab_corners[:, 0].min()
    a_max = Lab_corners[:, 1].max()
    a_min = Lab_corners[:, 1].min()
    b_max = Lab_corners[:, 2].max()
    b_min = Lab_corners[:, 2].min()

    print('\nLimits')
    print(f'L*:  {L_min:.4f}, {L_max:.4f})')
    print(f'a*:  {a_min:.4f}, {a_max:.4f})')
    print(f'b*:  {b_min:.4f}, {b_max:.4f})')

    return Lab_corners 

Regards,
Ram

**Comment by KelSolaar** [comment]
Hi @ramparvathaneni,

So what is happening is that the computations are done for D65 and the colourspace is chromatically adapted to this whitepoint. So the first numbers you have are correct but under D65.

However, as I looked at the code, I noticed that we are not passing the `illuminant` argument back to the `XYZ_to_Lab` definition, thus if you wanted to compute them for another whitepoint it would not work, so this behaviour is a defect.

An improvement would be to either not have the `illuminant` argument at all and only use that of the RGB colourspace, or have it default to that of the RGB colourspace.
